sheepdog/blueprint/routes/views/program/__init__.py

```diff
@@ -131,41 +122,98 @@ def create_project(program):
                 "name": "Burkitt Lymphoma Genome Sequencing Project",
                 "primary_site": "Lymph Nodes",
                 "dbgap_accession_number": "phs000527",
-                "state": "active"
+                "state": "active",
+                "dbgap_bypass_prefix": "test-",
+                "dbgap_bypass_range": 10
             }
     """
+
     auth.admin_auth()
-    doc = utils.parse.parse_request_json()
-    if not isinstance(doc, dict):
+    base_doc = utils.parse.parse_request_json()
+    if not isinstance(base_doc, dict):
         raise UserError('Program endpoint only supports single documents')
-    if doc.get('type') and doc.get('type') not in ['project']:
+
+    if base_doc.get('type') and base_doc.get('type') not in ['project']:
         raise UserError("Invalid post to program endpoint with type='{}'"
-                        .format(doc.get('type')))
+                        .format(base_doc.get('type')))
     # Parse project.code
-    project = doc.get('code')
+    project = base_doc.get('code')
     if not project:
         raise UserError("No project specified in key 'code'")
+
     project = project.encode('utf-8')
     # Parse dbgap accession number.
-    phsid = doc.get('dbgap_accession_number')
-    if not phsid:
-        raise UserError("No dbGaP accesion number specified.")
+    phsid = base_doc.get('dbgap_accession_number')
+
+    bypass_case_range = 0
+    if base_doc.get('bypass_case_range'):
+        bypass_case_range = base_doc['bypass_case_range']
+        # this dict will get used later for creating the project node
+        del base_doc['bypass_case_range']
+
+    bypass_case_prefix = 0
+    if base_doc.get('bypass_case_prefix'):
+        bypass_case_prefix = base_doc['bypass_case_prefix']
+        # this dict will get used later for creating the project node
+        del base_doc['bypass_case_prefix']
```

Comment:
Just to be cautious this doesn't delete the bypass_case_prefix right? I'm thinking python might be justcopying references.

Reply:
`base_doc` is the flask http request. Changing it is intentional.